### PR TITLE
HTCONDOR-1567-cmakery

### PR DESCRIPTION
**Pull request by GregThain**
Replace one call with unrolled calls to builtin add_subdirectory

[HTCONDOR-1567](https://opensciencegrid.atlassian.net/browse/HTCONDOR-1567)

<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [x] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed
- [x] Check for version history, if needed
- [x] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

[HTCONDOR-1567]: https://opensciencegrid.atlassian.net/browse/HTCONDOR-1567?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by Todd-L-Miller** [comment]
No need for documentation.

File: build/cmake/CondorConfigure.cmake [hunk 1]
```diff
@@ -504,9 +504,9 @@ if( NOT WINDOWS)
 	check_include_files("procfs.h" HAVE_PROCFS_H)
 	check_include_files("sys/procfs.h" HAVE_SYS_PROCFS_H)
 
-	check_type_exists("struct inotify_event" "sys/inotify.h" HAVE_INOTIFY)
-	check_type_exists("struct ifconf" "sys/socket.h;net/if.h" HAVE_STRUCT_IFCONF)
-	check_type_exists("struct ifreq" "sys/socket.h;net/if.h" HAVE_STRUCT_IFREQ)
+	check_struct_has_member("struct inotify_event" "len" "sys/inotify.h" HAVE_INOTIFY)
```
[the remark this answers is not in the document]
Comment (Todd-L-Miller) [reply build/cmake/CondorConfigure.cmake:507]: After in-person dicsussion -- GregT did in fact choose members that the code does in fact look at.  So it's incomplete but strictly better.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -16,15 +16,65 @@
...
-##########################################################################
-# adds all subtree elements which have a CMakeLists.txt file exclude
-# classads b/c we added earlier to elim externals dependency - all externals reference
-# one version of classads which is ${CONDOR_SOURCE_DIR}/src/classad
 
-# the 2nd arg is prio order for symbols ripping to prevent .o copy-mindging within the std:u
-add_subdirectories("${CONDOR_SOURCE_DIR}/src"
-	"${CONDOR_SOURCE_DIR}/src/bash_completion.d;${CONDOR_SOURCE_DIR}/src/classad_analysis;${CONDOR_SOURCE_DIR}/src/ccb;${CONDOR_SOURCE_DIR}/src/condor_io;${CONDOR_SOURCE_DIR}/src/condor_daemon_client;${CONDOR_SOURCE_DIR}/src/condor_daemon_core.V6;${CONDOR_SOURCE_DIR}/src/condor_procapi;${CONDOR_SOURCE_DIR}/src/condor_procd;${CONDOR_SOURCE_DIR}/src/condor_schedd.V6;${CONDOR_SOURCE_DIR}/src/condor_sysapi;${CONDOR_SOURCE_DIR}/src/condor_utils"
-	"${CONDOR_SOURCE_DIR}/src/classad;${CONDOR_SOURCE_DIR}/src/safefile")
+add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/annex)
```
[the remark this answers is not in the document]
Comment (Todd-L-Miller) [reply src/CMakeLists.txt:20]: GregT said he'd test this in an out-of-source build.  (Doesn't the BaTLab do out-of-source builds?  If not, why not?)

[the remark this answers is not in the document]
Comment (GregThain) [reply src/CMakeLists.txt]: Looks like it does work.  pushed fix.

**Review by Todd-L-Miller — APPROVED** [review]

**Pull request merged**